File: expense-report/python/src/expense_report/spending_policy.py
from .category import Category
from .money import Money

_PER_ITEM_LIMITS = {
    Category.MEALS: Money(50),
    Category.TRAVEL: Money(500),
    Category.ACCOMMODATION: Money(200),
    Category.EQUIPMENT: Money(1000),
    Category.OTHER: Money(100),
}

REPORT_MAXIMUM = Money(5000)
APPROVAL_THRESHOLD = Money(2500)


def limit_for(category: Category) -> Money:
    return _PER_ITEM_LIMITS[category]

File: expense-report/csharp/src/ExpenseReport/SpendingPolicy.cs
namespace ExpenseReport;

public static class SpendingPolicy
{
    private static readonly Dictionary<Category, Money> PerItemLimits = new()
    {
        { Category.Meals, new Money(50m) },
        { Category.Travel, new Money(500m) },
        { Category.Accommodation, new Money(200m) },
        { Category.Equipment, new Money(1000m) },
        { Category.Other, new Money(100m) },
    };

    public static readonly Money ReportMaximum = new(5000m);
    public static readonly Money ApprovalThreshold = new(2500m);

    public static Money LimitFor(Category category) => PerItemLimits[category];
}
